src/multirw.rs

```diff
@@ -0,0 +1,224 @@
+        let mut cursor = self.tx.open_ro_cursor(store.0).map_err(
+            StoreError::LmdbError,
+        )?;
+        let iter = cursor.iter_dup_of(k);
+        //Ok(Iter{ iter, cursor })
+        Ok(Iter { iter })
+    }
+
+    /// Consume this MultiCursor and give the `MultiWriter` back
+    /// So that it may perform additional tasks
+    pub fn into_writer(self) -> MultiWriter<'env, K> {
+        MultiWriter {
+            tx: self.tx,
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<'env, K> MultiWriter<'env, K>
+where
+    K: AsRef<[u8]>,
+{
+    pub(crate) fn new(txn: RwTransaction) -> MultiWriter<K> {
+        MultiWriter {
+            tx: txn,
+            phantom: PhantomData,
+        }
+    }
+
+    /// This cursor consumes the writer, as it is not safe to attempt to access records while using
+    /// put or delete, as the location of the records might change
+    pub fn into_cursor(self) -> MultiCursor<'env, K> {
+        MultiCursor {
+            tx: self.tx,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Insert a value at the specified key.
+    /// This put will allow duplicate entries.  If you wish to have duplicate entries
+    /// rejected, use the `put_flags` function and specify NO_DUP_DATA
+    pub fn put(&mut self, store: MultiStore, k: K, v: &Value) -> Result<(), StoreError> {
+        // TODO: don't allocate twice.
+        let bytes = v.to_bytes()?;
+        self.tx
+            .put(store.0, &k, &bytes, WriteFlags::empty())
+            .map_err(StoreError::LmdbError)
+    }
+
+    pub fn put_with_flags(&mut self, store: MultiStore, k: K, v: &Value, flags: WriteFlags) -> Result<(), StoreError> {
+        // TODO: don't allocate twice.
+        let bytes = v.to_bytes()?;
+        self.tx.put(store.0, &k, &bytes, flags).map_err(
+            StoreError::LmdbError,
+        )
+    }
+
+    pub fn delete(&mut self, store: MultiStore, k: K, v: &Value) -> Result<(), StoreError> {
+        self.tx.del(store.0, &k, Some(&v.to_bytes()?)).map_err(
+            StoreError::LmdbError,
+        )
+    }
+
+    pub fn commit(self) -> Result<(), StoreError> {
+        self.tx.commit().map_err(StoreError::LmdbError)
+    }
+
+    pub fn abort(self) {
+        self.tx.abort();
+    }
+}
+
+impl<'env, K> MultiReader<'env, K>
+where
+    K: AsRef<[u8]>,
+{
+    pub(crate) fn new(txn: RoTransaction) -> MultiReader<K> {
+        MultiReader {
+            tx: txn,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Provides a cursor to all of the values for the duplicate entries that match this key
+    pub fn get(&self, store: MultiStore, k: K) -> Result<Iter, StoreError> {
+        let mut cursor = self.tx.open_ro_cursor(store.0).map_err(
+            StoreError::LmdbError,
+        )?;
+        let iter = cursor.iter_dup_of(k);
+        //Ok(MultiIter { iter, cursor, })
+        Ok(Iter { iter })
+    }
+
+    /// Cancel this read transaction (not particularly useful)
```

Comment:
The LMDB [Getting Started](http://www.lmdb.tech/doc/starting.html) page talks about the utility of aborting read transactions:

> For read-only transactions, obviously there is nothing to commit to storage. The transaction still must eventually be aborted to close any database handle(s) opened in it, or committed to keep the database handles around for reuse in new transactions.
>
> In addition, as long as a transaction is open, a consistent view of the database is kept alive, which requires storage. A read-only transaction that no longer requires this consistent view should be terminated (committed or aborted) when the view is no longer needed (but see below for an optimization).

Reply:
rgr that.  I'll remove the contents in the parens :)